packages/medusa-plugin-typesense/src/utils/transformer.ts

```diff
@@ -0,0 +1,43 @@
+import { variantKeys } from "@medusajs/types"
+
+const prefix = `variant`
+
+export const transformProduct = (product: any) => {
+  let transformedProduct = { ...product } as Record<string, unknown>
+
+  const initialObj = variantKeys.reduce((obj, key) => {
+    obj[`${prefix}_${key}`] = []
+    return obj
+  }, {})
+  initialObj[`${prefix}_options_value`] = []
+
+  const flattenedVariantFields = product.variants.reduce((obj, variant) => {
+    variantKeys.forEach((k) => {
+      if (k === "options" && variant[k]) {
```

Comment:
should we use `variant.options` here for clarity?

Reply:
Not sure I understand

Reply:
instead of `variant[k]` 😄 or not necessarily here but in the subsequent lines since we already validate that we are handling options here

Reply:
very nit btw